Doc/faq/programming.rst

```diff
@@ -1334,11 +1340,12 @@ that even though there was an error, the append worked::
     ['foo', 'item']
 
 To see why this happens, you need to know that (a) if an object implements an
-``__iadd__`` magic method, it gets called when the ``+=`` augmented assignment
+:meth:`~object.__iadd__` magic method, it gets called when the ``+=`` augmented
+assignment
 is executed, and its return value is what gets used in the assignment statement;
-and (b) for lists, ``__iadd__`` is equivalent to calling ``extend`` on the list
+and (b) for lists, :meth:`__iadd__` is equivalent to calling :meth:`list.extend` on the list
```

Comment:
Since we don't want the first reference to resolve (since it was already linked above), add `!` here. As a sidenote, [`~!object.__attr__` doesn't actually work as we thought](https://github.com/python/docs-community/issues/52#issuecomment-1278268414), so you do have to specify only the part(s) of the name you actually want to be displayed there.

```suggestion
and (b) for lists, :meth:`!__iadd__` is equivalent to calling :meth:`list.extend` on the list
```

Right now there is no linkable target for `list.extend` (or even for the abstract `extend` method), but that is a valid warning that should be fixed by refactoring the documentation, as this also causes problems for search, indexing and other things. See #86272 where I discuss this further. I've been planning on implementing that, but just haven't gotten to it yet. Therefore, I propose we leave it, so it will be automatically resolved if and when that is added and in the meantime is a reminder of the problem.

Reply:
By leave it, do you mean leaving it as literal or as `` :meth:`~list.extend` ``?

Reply:
Sorry, I meant leaving it as it is, `` `:meth:`list.extend` ``, though you make a good point that :meth:`~list.extend` would make more sense in this context (since `list` is mentioned a couple words later). I've updated the suggestion accordingly, and will check for any others.